src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1000,6 +1006,10 @@ private static Type substituteTypeVariables(final Type type, final Map<TypeVaria
      * @param parameterizedType the parameterized type
      * @param typeVarAssigns the map to be filled
      */
+    @SuppressWarnings({"assignment.type.incompatible","compound.assignment.type.incompatible", "array.access.unsafe.high"}) /*
+    #5 - getRawType(parameterizedType).getTypeParameters() has same length as parameterizedType.getActualTypeArguments()
```

Comment:
This is always true, right? Could you add an annotation on the type that `getRawType` returns to indicate that?

Reply:
Actually, it's not a property of `getRawType(parameterizedType)`, but of `getRawType(parameterizedType).getTypeParameters()`, so I can't annotate it. Do you have any ideas to achieve so?

Reply:
You are right - I misread them as being the same object.